src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_secrets.rs

```diff
@@ -37,6 +52,189 @@ pub fn is_base64_url_encoding_char(ch: char) -> bool
                    ch == '_';
 }
 
+/// Generate a u64 an HIS v1 compliant checksum seed from a string literal
+/// that is 8 characters long and ends with at least one digit, e.g., 'ReadKey0', 'RWSeed00',
+/// etc. The checksum seed is used to initialize the Marvin32 algorithm to watermark a
+/// specific class of generated security keys.
+///
+/// # Arguments
+///
+/// * `versioned_key_kind` - A readable name that identifies a specific set of generated keys with at least one trailing digit in the name.
+///
+/// # Returns
+///
+/// The computed checksum seed as a u64.
+///
+/// # Errors
+///
+/// This function will return an error if the `versioned_key_kind` does not meet the required criteria.
+pub fn compute_his_v1_checksum_seed(versioned_key_kind: &str) -> u64 {
+
+    if versioned_key_kind.len() != 8 || !versioned_key_kind.chars().nth(7).unwrap().is_digit(10) {
+        panic!("The versioned literal must be 8 characters long and end with a digit.");
+    }
+
+    let bytes = versioned_key_kind.as_bytes().iter().rev().cloned().collect::<Vec<u8>>();
+    let result = u64::from_le_bytes(bytes.try_into().unwrap());
+
+    result
+}
+
+pub fn try_validate_common_annotated_key(key: &str, base64_encoded_signature: &str) -> bool {
```

Comment:
@michaelcfanning In the validator, I assert a 63 byte key (I can relax this to 64 bytes if needed, but the generator does not produce it yet). Then the first 3 bytes of Marvin is checked.